ledger/apptxn_test.go

```diff
@@ -624,3 +631,264 @@ func TestRekeyActionCloseAccount(t *testing.T) {
+         int pay
+         tx_field TypeEnum
+         int 5000
+         tx_field Amount
+         txn Accounts 1
+         tx_field Receiver
+         tx_submit
+`),
+	}
+
+	fund := txntest.Txn{
+		Type:     "pay",
+		Sender:   addrs[0],
+		Receiver: basics.AppIndex(1).Address(),
+		Amount:   200000, // account min balance, plus fees
+	}
+
+	payout1 := txntest.Txn{
+		Type:          "appl",
+		Sender:        addrs[1],
+		ApplicationID: basics.AppIndex(1),
+		Accounts:      []basics.Address{addrs[1]}, // pay self
+	}
+
+	eval := l.nextBlock(t)
+	eval.txns(t, &create, &fund)
+	vb := l.endBlock(t, eval)
+	require.Equal(t, 2, int(vb.blk.TxnCounter))
+
+	eval = l.nextBlock(t)
+	eval.txns(t, &payout1)
+	vb = l.endBlock(t, eval)
+	require.Equal(t, 4, int(vb.blk.TxnCounter))
+}
+
+// TestAcfgAction ensures assets can be created and configured in teal
+func TestAcfgAction(t *testing.T) {
+	partitiontest.PartitionTest(t)
+
+	genBalances, addrs, _ := newTestGenesis()
+	l := newTestLedger(t, genBalances)
+	defer l.Close()
+
+	appIndex := basics.AppIndex(1)
+	app := txntest.Txn{
+		Type:   "appl",
+		Sender: addrs[0],
+		ApprovalProgram: main(`
+         tx_begin
+         int acfg
+         tx_field TypeEnum
+
+         txn ApplicationArgs 0
+         byte "create"
+         ==
+         bz manager
+		 int 1000000
+		 tx_field ConfigAssetTotal
+		 int 3
+		 tx_field ConfigAssetDecimals
+		 byte "oz"
+		 tx_field ConfigAssetUnitName
+		 byte "Gold"
+		 tx_field ConfigAssetName
+		 byte "https:://gold.rush/"
+		 tx_field ConfigAssetURL
+		 global CurrentApplicationAddress
+         dup
+         dup2
+		 tx_field ConfigAssetManager
+		 tx_field ConfigAssetReserve
+		 tx_field ConfigAssetFreeze
+		 tx_field ConfigAssetClawback
+         b submit
+manager:
+         txn ApplicationArgs 0
+         byte "manager"
+         ==
+         bz reserve
+reserve:
+         txn ApplicationArgs 0
+         byte "reserve"
+         ==
+         bz freeze
+freeze:
+         txn ApplicationArgs 0
+         byte "freeze"
+         ==
+         bz clawback
+clawback:
+         txn ApplicationArgs 0
+         byte "manager"
+         ==
+submit:  tx_submit
+`),
+	}
+
+	fund := txntest.Txn{
+		Type:     "pay",
+		Sender:   addrs[0],
+		Receiver: appIndex.Address(),
+		Amount:   200000, // exactly account min balance + one asset
+	}
+
+	eval := l.nextBlock(t)
+	eval.txns(t, &app, &fund)
+	l.endBlock(t, eval)
+
+	create_asa := txntest.Txn{
+		Type:            "appl",
+		Sender:          addrs[1],
+		ApplicationID:   appIndex,
+		ApplicationArgs: [][]byte{[]byte("create")},
+	}
+
+	eval = l.nextBlock(t)
+	// Can't create an asset if you have exactly 200,000 and need to pay fee
+	eval.txn(t, &create_asa, "balance 199000 below min 200000")
+	// fund it some more and try again
+	eval.txns(t, fund.Noted("more!"), &create_asa)
+	vb := l.endBlock(t, eval)
+
+	asaIndex := vb.blk.Payset[1].EvalDelta.InnerTxns[0].ConfigAsset
+	require.Equal(t, basics.AssetIndex(5), asaIndex)
```

Comment:
Could you also pull out the params of the newly created asset and check its fields match the values set by the AVM?

Also, I suppose it's not possible to actually transfer the newly created asset or use the `asset_param_get` op on it since it can't be in the foreign assets array? I wonder if we should make an exception in this case and allow the asset to be "available" anyway.

Reply:
WiIll do, on testing.

I was strongly considering adding a special case (as we did for the app address) that would allow the use of an asset that had been created in the same transaction (or txgroup?). But as for as I can tell, it's pretty much impossible to use, because no recipient could possibly be opted-in yet.

Reply:
Well theoretically the app could have another account rekeyed to it, so it could potentially opt that account into the new asset and perform a transfer.

The bigger issue I see is that there's currently no way to get the ID of the created asset. If we don't support this, there's no reason to support the special case of adding newly created assets to the "available" list.

Reply:
I'd certainly like it too.  Let me see if I can think of a decent interface that doesn't paint us into a corner.  Ideas welcome.

Reply:
Perhaps
```
itxna Logs X
itxn ConfigAsset
```
(and attendant `itxn NumLogs`, `itxnas`)

These would all return results from previous `tx_submit` execution.  And perhaps they become `itxn_begin` etc.